docs/information/user_extensions.md

```diff
@@ -0,0 +1,37 @@
+# User extensions
+
+User extensions is a way to extend zigbee2mqtt behaviour, user extensions works in same way as internal
+
+To get familiar with  extensions framework please read (source code)[https://github.com/Koenkk/zigbee2mqtt/tree/master/lib/extension]
+
+User extensions are stored in `data/extensions` folder
+
+User extension has to export js Class or Function
+
+example:
+
+`data/extensions/my-first-extension.js`
+
+```js
+class MyExampleExtension {
+    constructor(zigbee, mqtt, state, publishEntityState, eventBus, settings, logger) {
+        logger.info('Loaded  MyExampleExtension');
+        mqtt.publish('example/extension', 'hello from MyExampleExtension')
+    }
+}
+
+module.exports = MyExampleExtension;
+```
+
+## Extensions list
+
+Once zigbee2mqtt starts it publishes `zigbee2mqtt/bridge/extensions` with payload `["my-first-extension.js"]`
```

Comment:
For future extendability, should we change `["my-first-extension.js"]` to `[{"name": "my-first-extension.js"}]`?

Food for thought: maybe we can also add a `content` property, this would eliminate the `zigbee2mqtt/bridge/extension/request/read` request.

Reply:
makes sense, also it will simplify frontend code as well

Reply:
updated